docs/explanation/architectures.rst

```diff
@@ -0,0 +1,168 @@
+Architectures
+=============
+
+Build-on, build-for, and run-on
+-------------------------------
+
+``build-on``, ``build-for``, and ``run-on`` are used inside the root
+``architectures`` keyword in a ``snapcraft.yaml``.
+
+* ``core22`` uses ``build-on`` and ``build-for``
+* ``core20`` uses ``build-on`` and ``run-on``
+
+build-on
+^^^^^^^^
+
+The architecture of the machine that the snap is built on (i.e. the machine
+where snapcraft runs).
+
+build-for
+^^^^^^^^^
+
+The architecture of the machine that the snap is built for (i.e. the machine
+where the snap will be installed).
+
+run-on
+^^^^^^
+
+This is used for ``core20`` and has the same meaning as ``build-for``. It has
+been replaced with the preferred term ``build-for`` in ``core22``.
+
+.. _build-plans:
+
+Build plans
+-----------
+
+A build plan is a list of what snaps snapcraft will build. It can be defined
+in the ``snapcraft.yaml`` and filtered with command-line arguments and
+environment variables.
+
+core22
+^^^^^^
+
+Snapcraft can create multiple snaps, each built for a different architecture.
+Consider the following ``snapcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: amd64
+    - build-on: [amd64, arm64]
+      build-for: arm64
+
+If snapcraft executes on an ``amd64`` machine, then it will create the
+following build plan:
```

Comment:
What happened if executes on an ``arm64`` machine? we may say it only get ``arm64`` snap.

Reply:
Good call, updated.